app/models/manageiq/providers/hawkular/middleware_manager/middleware_domain.rb

```diff
@@ -1,4 +1,14 @@
 module ManageIQ::Providers
   class Hawkular::MiddlewareManager::MiddlewareDomain < MiddlewareDomain
+    AVAIL_TYPE_ID = 'Domain%20Availability~Domain%20Availability'.freeze
+
+    def availability
+      properties['Availability']
+    end
+
+    def availability=(value)
+      self.properties ||= {}
```

Comment:
This could be at the `#properties` method.

Reply:
Totally right!